application/src/org/commcare/entity/CommCareEntity.java

```diff
@@ -165,8 +186,42 @@ protected void loadEntity(TreeReference element) {
         loadTexts(ec);
     }
 
+    /**
+     * Given a detail, return an array of details that will contain either
+     * - all child details
+     * - a single-element array containing the given detail, if it has no children
+     * @param d The detail to flatten.
+     */
+    private Detail[] getFlattenedDetails(Detail d) {
```

Comment:
Can we move this to the detail class or somewhere that will make it possible for us to unify the behavior on platforms which don't support tabs?